apps/s_client.c

```diff
@@ -1742,7 +1762,11 @@ int s_client_main(int argc, char **argv)
     if (fallback_scsv)
         SSL_set_mode(con, SSL_MODE_SEND_FALLBACK_SCSV);
 
-    if (servername != NULL) {
+    if (!noservername && (servername != NULL || dane_tlsa_domain == NULL)) {
+        if (servername == NULL)
+            servername = host;
+        if (servername == NULL)
+            servername = "localhost";
```

Comment:
That perhaps makes some sense, but will likely involve further changes to the tests and docs.  If we implicitly connect to "localhost" however, then a default connect host should perhaps not behave differently from an explicit connect to "localhost".